src/NServiceBus.Core/Pipeline/PipelineExecutionExtensions.cs

```diff
@@ -64,29 +67,41 @@ public static Delegate CreatePipelineExecutionExpression(this IBehavior[] behavi
         }
 
         /// <code>
-        /// context{i} => ((BehaviorType)context{i}.Extensions.Behaviors[i]).Invoke(context{i+1} => previous)
+        /// context{i} => GetBehavior(context{i}, {i}).Invoke(context{i+1} => previous)
         /// </code>>
         static Delegate CreateBehaviorCallDelegate(MethodInfo methodInfo, ParameterExpression outerContextParam, Type behaviorType, Delegate previous, int i, List<Expression> expressions = null)
         {
-            PropertyInfo extensionProperty = typeof(IExtendable).GetProperty("Extensions");
-            Expression extensionPropertyExpression = Expression.Property(outerContextParam, extensionProperty);
-            PropertyInfo behaviorsProperty = typeof(ContextBag).GetProperty("Behaviors", BindingFlags.Instance | BindingFlags.NonPublic);
-            Expression behaviorsPropertyExpression = Expression.Property(extensionPropertyExpression, behaviorsProperty);
-            Expression indexerPropertyExpression = Expression.ArrayIndex(behaviorsPropertyExpression, Expression.Constant(i));
-            Expression castToBehavior = Expression.Convert(indexerPropertyExpression, behaviorType);
-            Expression body = Expression.Call(castToBehavior, methodInfo, outerContextParam, Expression.Constant(previous));
+            MethodInfo getBehaviorMethodInfo = typeof(PipelineExecutionExtensions).GetMethod("GetBehavior")!.MakeGenericMethod(outerContextParam.Type, behaviorType);
```

Comment:
Could `typeof(PipelineExecutionExtensions).GetMethod("GetBehavior")` be a static and potentially speed up startup? (Perhaps negligibly)

Reply:
Yeah that is totally doable. Pushed.